frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -214,10 +214,13 @@ fn construct_runtime_final_expansion(
 	let ExplicitRuntimeDeclaration {
 		name,
 		where_section: WhereSection { block, node_block, unchecked_extrinsic },
-		pallets,
+		mut pallets,
 		pallets_token,
 	} = definition;
 
+	// Ensure that the code is generated based on the pallet index ordering:
+	pallets.sort_by_key(|p| p.index);
```

Comment:
Curious what happens to that when there are no explicitly supplied indices. Does the macro generate indices based on the order of declarations?

Reply:
it does, so if you set 10 and then later 5, then it would increment up and clash but it would throw an error saying that there are 2 of the same pallet index. It seems reasonable behavior.